File: src/data/UserTicket.ts
interface UserTicket
{
    Id: string,
    UserId: string,
    TicketId: string
}

export default UserTicket

/*
        [Key]
        public string Id {  get; set; }

        // Foreign key
        public string UserId { get; set; }

        // Foreign key
        public string TicketId { get; set; }

        [ForeignKey("UserId")]
        public User User {  get; set; }

        [ForeignKey("TicketId")]
        public Ticket Ticket {  get; set; }
*/

File: Backend/Data/Model/UserTicket.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Data.Model
{
    // M-to-N relationship between User and Ticket table
    public class UserTicket
    {
        // Foreign key (composite)
        [ForeignKey("User")]
        public string UserId { get; set; }

        // Foreign key (composite)
        [ForeignKey("Ticket")]
        public string TicketId { get; set; }

        // Navigation properties for tables
        public User User {  get; set; }
        public Ticket Ticket {  get; set; }
    }
}
